library/src/main/kotlin/com/connectrpc/protocols/GRPCInterceptor.kt

```diff
@@ -161,15 +163,22 @@ internal class GRPCInterceptor(
                     onCompletion = { result ->
                         val trailers = result.trailers
                         val completion = completionParser.parse(emptyMap(), trailers)
-                        val connectError = grpcCompletionToConnectError(completion, serializationStrategy, result.error)
-                        StreamResult.Complete(
-                            code = connectError?.code ?: Code.OK,
-                            error = connectError,
-                            trailers = trailers,
-                        )
+                        if (completion != null) {
+                            val exception = completion.toConnectExceptionOrNull(
+                                serializationStrategy,
+                                result.cause,
+                            )
+                            StreamResult.Complete(
+                                code = exception?.code ?: Code.OK,
+                                cause = exception,
+                                trailers = trailers,
+                            )
+                        } else {
+                            val exception = ConnectException(result.code)
+                            StreamResult.Complete(exception.code, exception, trailers)
```

Comment:
We may be losing an underlying cause here - so we should just return the result here.

Reply:
I worked around the problem by using the result's code property value in the connect exception constructor, but I like that idea better.

e83429aa067fa83cda356955328eb0d0f96bc0ec